build/templates/_library_interpreter.py/numpy_read_method.py.mako

```diff
@@ -0,0 +1,50 @@
+<%page args="f, config, method_template"/>\
+<%
+    '''Renders a Intermediate method for reading into a numpy.array corresponding to the passed-in function metadata.'''
+
+    import build.helper as helper
+
+    parameters = f['parameters']
+    param_names_method = helper.get_params_snippet(f, helper.ParameterUsageOptions.LIBRARY_INTERPRETER_NUMPY_INTO_METHOD_DECLARATION)
+    param_names_library = helper.get_params_snippet(f, helper.ParameterUsageOptions.LIBRARY_METHOD_CALL)
+    param_ctypes_library = helper.get_params_snippet(f, helper.ParameterUsageOptions.CTYPES_ARGTYPES)
+
+    ivi_dance_parameters = helper.filter_ivi_dance_parameters(f)
+    ivi_dance_size_parameter = helper.find_size_parameter(ivi_dance_parameters, parameters)
+    len_parameters = helper.filter_len_parameters(f)
+    len_size_parameter = helper.find_size_parameter(len_parameters, parameters)
+    assert ivi_dance_size_parameter is None or len_size_parameter is None, str(f)
```

Comment:
I'm guessing this `, str(f)` was also for debugging?

Reply:
Same here. Removed the ivi_dance code from the numpy functions.